ui/dashboards/src/components/Variables/VariableEditor.tsx

```diff
@@ -260,6 +267,54 @@ export function VariableEditor(props: {
                 </Button>
               </Box>
             </Stack>
+
+            {externalVariableDefinitions.map((extVar) => (
+              <Stack key={extVar.source}>
+                <Typography variant="h2">{capitalize(extVar.source)} Variables</Typography>
+                <TableContainer>
+                  <Table sx={{ minWidth: 650 }} aria-label="table of variables">
+                    <TableHead>
+                      <TableRow>
+                        <TableCell>Visibility</TableCell>
+                        <TableCell>Name</TableCell>
+                        <TableCell>Type</TableCell>
+                        <TableCell align="right" />
+                      </TableRow>
+                    </TableHead>
+                    <TableBody>
+                      {extVar.definitions.map((v) => (
+                        <TableRow key={v.spec.name}>
+                          <TableCell component="th" scope="row">
+                            <Switch checked={v.spec.display?.hidden !== true} disabled />
```

Comment:
I'm confused on the expected behavior here, it makes senses that you can't edit the project variables or remove them when on a dashboard so they are disabled. But how do I hide them on a dashboard if the 'Visibility' toggle is disabled? I created a bunch of test project variables and want to use some of them on a dashboard but not necessarily show them all or else there are way too many variables in the TemplateVariableList in the dashboard toolbar. 

Am I missing a way to configure `spec.display.hidden` for project variables in the UI somewhere? Screen recording with example:

https://github.com/perses/perses/assets/9369625/83cd21eb-becb-4da2-8b7a-d21e11024980

Reply:
Oh I missed the other PR #1290, might need to test with that one too locally? Since these changes are all related and global variables CRUD is coming too, would it make sense to merge these PRs into a feature branch first instead of going straight into `main`, I just created this [feature/global-and-project-variables](https://github.com/perses/perses/tree/feature/global-and-project-variables) branch where we could potentially consolidate and test these changes together before they go into main (cc @AntoineThebaud too if any thoughts?).

Reply:
Hello, the way to hide an external variable for now is to create a local variable with same name and set it to hidden (even with other PR merged) but I confess it may sound to be a bit tricky. 

We could imagine a more straightforward way to hide it, I'm personally fine with the current way but if you have a better idea I can implement it. 

Regarding the project var crud PR, it's been already merged actually ^^ Except if you cherry picked the commits to remove them from main, it's already on main branch. Then I'm not sure the feature branch is relevant here WDYT ?

Reply:
@sjcobb Yes this the expected behavior for now, knowing that you always have the solution @celian-garcia described to hide the variable. Still, I agree that this is something that we should expect our future users to ask and I personally already had it in my personal todo (_"Study/submit proposal: editable visiblity for external variables on dashboards"_). 

Actually I would like to go even further and let people chose to "import" or not a given external variable (since with the visibility thing you already have the need to store a reference to the external variables in each dashboard individually, I don't expect that this would be more costly to implement). This way the external variables would be some kind of libraries that you'd decide to use or not.

It's probably worth creating a new Discussion for this topic. Anyway I think for now we should go with this first implem.

NB: just found the request for similar feature on Grafana side https://github.com/grafana/grafana/issues/1959, we may convert quite some people to Perses with our external vars ;)

Reply:
Regarding the bug you spotted @sjcobb, let me rephrase from what I understood after our discussion and some quick investigation 

There are two bugs coming from the same situation : when you make a local variable (say instance) use an external variable (say job or "jobFromProj")  
- Bug one : you can´t save the dashboard as the backend make a validation to check the existence of the variable used
=> I will improve the perses backend to take in account global and project vars
- Bug two : a no data message appears on the panels using $instance.
=> I will check the depends on mechanism which is proper to the front to see if I didn't miss to include the external vars.

Thanks!

Reply:
Yeah that's a good summary and thanks for fixing the BE! 
For Bug two, I'm not 100% sure why the charts were showing 'No Data' in that case, but we'll definitely need to test charts where dashboard variables depend on project variables but the project variables take longer to load--same for when globals are added.

Put a summary of the idea we discussed on Mon in notes [here](https://github.com/perses/perses/discussions/947?sort=new#discussioncomment-6469469), in the future it'd be nice to look into that, basically a quicker way to define the lower level variable with the correct `spec.display` where the user just clicks the toggle and the UI takes care of the mapping under the hood

Reply:
I am currently looking into the second bug yeah, I'll find out (hopefully :face_with_spiral_eyes: )

For the display hidden, this is a good idea to have a shortcut, but putting this on the toggle would be in my opinion a bit too confusing for the user. Maybe a button at the very right "override locally" ? that would just create the local variable, and then people would just have to toggle the local var. I'm sure I can quickly do a proto to make you play with it and see if you find it better than what we have here.

Reply:
Here is the issue I created https://github.com/perses/perses/issues/1312

Reply:
Regarding the override shortcut, I pushed a proposal with a gif in the description.

I close this conv as resolved as everything is answered.